app/controllers/tradingpremises/IsResidentialController.scala

```diff
@@ -57,7 +57,7 @@ class  IsResidentialController @Inject()(
               address <- getAddress(bm) if isFirstTradingPremises(tradingPremises, index)
             } yield address
```

Comment:
This constructing of the `address` val could be removed completely as it's no longer used.